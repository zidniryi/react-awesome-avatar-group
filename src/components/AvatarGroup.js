import React from 'react';

const AvatarGroup = ({ maxLength = 3, size = 'md', names }) => {
    const getInitials = (name) => {
        const words = name.split(' ');
        return words.map((word) => word.charAt(0).toUpperCase()).join('').substring(0, 2);
    };

    const renderAvatars = () => {
        const remainingCount = names.length - maxLength;
        const avatarsToRender = names.slice(0, maxLength);
        const avatars = avatarsToRender.map(({ name, image }, index) => {
            const initials = getInitials(name);
            const avatarStyle = {
                marginRight: "-15px",
                border: "3px solid white",
                marginLeft: "8%",
                display: 'flex',
                justifyContent: 'center',
                alignItems: 'center',
                width: size === 'xs' ? '24px' : size === 'sm' ? '32px' : size === 'md' ? '40px' : '48px',
                height: size === 'xs' ? '24px' : size === 'sm' ? '32px' : size === 'md' ? '40px' : '48px',
                borderRadius: '50%',
                backgroundColor: '#E6EAF4',
                color: '#767B87',
                fontSize: size === 'xs' ? '12px' : size === 'sm' ? '16px' : size === 'md' ? '20px' : '24px',
            };

            if (image) {
                avatarStyle.backgroundImage = `url(${image})`;
                avatarStyle.backgroundSize = 'contain';
                avatarStyle.backgroundRepeat = 'no-repeat';
                avatarStyle.color = "transparent"
            }

            return (
                <div key={`avatar-${index}`} style={{ display: 'inline-block' }} data-testid="avatar-initial" >
                    <div
                        style={avatarStyle}
                        data-testid="avatar-initial-text"
                    >
                        {initials}
                    </div>
                </div>
            );
        });

        if (remainingCount > 0) {
            avatars.push(
                <div key="avatar-remaining"
                    data-testid="avatar-initial-text"
                    style={{ display: 'inline-block' }}>
                    <div
                        data-testid="avatar-initial-item"
                        style={{
                            marginRight: "-15px",
                            border: "3px solid white",
                            display: 'flex',
                            justifyContent: 'center',
                            alignItems: 'center',
                            width: size === 'xs' ? '24px' : size === 'sm' ? '32px' : size === 'md' ? '40px' : '48px',
                            height: size === 'xs' ? '24px' : size === 'sm' ? '32px' : size === 'md' ? '40px' : '48px',
                            borderRadius: '50%',
                            backgroundColor: '#E6EAF4',
                            color: '#767B87',
                            fontSize: size === 'xs' ? '12px' : size === 'sm' ? '16px' : size === 'md' ? '20px' : '24px',
                        }}
                    >
                        +{remainingCount}
                    </div>
                </div>
            );
        }

        return avatars;
    };

    return <div>{renderAvatars()}</div>;
};

export default AvatarGroup;