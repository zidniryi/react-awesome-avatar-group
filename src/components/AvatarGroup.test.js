import React from 'react';
import { render } from '@testing-library/react';
import AvatarGroup from './AvatarGroup';

describe('AvatarGroup', () => {
    const names = [
        { name: 'John Doe', image: 'https://www.w3schools.com/html/pic_trulli.jpg' },
        { name: 'Jane Smith', image: '' },
        { name: 'Mike Johnson', image: 'https://www.w3schools.com/html/img_chania.jpg' },
        { name: 'Sarah Lee', image: 'https://www.w3schools.com/html/img_girl.jpg' },
        { name: 'Tom Davis', image: '' },
        { name: 'Emily White', image: 'https://www.w3schools.com/html/img_lights.jpg' },
    ];

    it('should render properly with default props', () => {
        const { container } = render(<AvatarGroup names={names} />);
        expect(container.firstChild).toMatchSnapshot();
    });

    it('should render avatars with correct size', () => {
        const { getAllByTestId } = render(<AvatarGroup names={names} size="sm" />);
        const avatars = getAllByTestId('avatar-initial-text');
        expect(avatars[0]).toHaveStyle({ width: '32px', height: '32px', fontSize: '16px' });
    });

    it('should render avatars with remaining count', () => {
        const { getByTestId } = render(<AvatarGroup names={names} maxLength={2} />);
        const remainingAvatar = getByTestId('avatar-initial-item');
        expect(remainingAvatar).toHaveTextContent('+4');
    });
});