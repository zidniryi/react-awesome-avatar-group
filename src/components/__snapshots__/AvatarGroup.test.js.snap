// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AvatarGroup should render properly with default props 1`] = `
<div>
  <div
    data-testid="avatar-initial"
    style="display: inline-block;"
  >
    <div
      data-testid="avatar-initial-text"
      style="margin-right: -15px; border: 3px solid white; margin-left: 8%; display: flex; justify-content: center; align-items: center; width: 40px; height: 40px; border-radius: 50%; background-color: rgb(230, 234, 244); color: transparent; font-size: 20px; background-image: url(https://www.w3schools.com/html/pic_trulli.jpg); background-size: contain; background-repeat: no-repeat;"
    >
      JD
    </div>
  </div>
  <div
    data-testid="avatar-initial"
    style="display: inline-block;"
  >
    <div
      data-testid="avatar-initial-text"
      style="margin-right: -15px; border: 3px solid white; margin-left: 8%; display: flex; justify-content: center; align-items: center; width: 40px; height: 40px; border-radius: 50%; background-color: rgb(230, 234, 244); color: rgb(118, 123, 135); font-size: 20px;"
    >
      JS
    </div>
  </div>
  <div
    data-testid="avatar-initial"
    style="display: inline-block;"
  >
    <div
      data-testid="avatar-initial-text"
      style="margin-right: -15px; border: 3px solid white; margin-left: 8%; display: flex; justify-content: center; align-items: center; width: 40px; height: 40px; border-radius: 50%; background-color: rgb(230, 234, 244); color: transparent; font-size: 20px; background-image: url(https://www.w3schools.com/html/img_chania.jpg); background-size: contain; background-repeat: no-repeat;"
    >
      MJ
    </div>
  </div>
  <div
    data-testid="avatar-initial-text"
    style="display: inline-block;"
  >
    <div
      data-testid="avatar-initial-item"
      style="margin-right: -15px; border: 3px solid white; display: flex; justify-content: center; align-items: center; width: 40px; height: 40px; border-radius: 50%; background-color: rgb(230, 234, 244); color: rgb(118, 123, 135); font-size: 20px;"
    >
      +
      3
    </div>
  </div>
</div>
`;
